$NetBSD: patch-src_CMakeLists.txt,v 1.4 2020/08/11 18:06:10 gdt Exp $

* hunk 1

On NetBSD, don't use -ldl

Sent upstream via email 20190427, again on 20200811, and applied
upstream on
20200811.

* hunk 2

\todo Explain why this hunk is here.

\todo File hunk upstream.

--- src/CMakeLists.txt.orig	2020-08-11 11:18:06.000000000 +0000
+++ src/CMakeLists.txt
@@ -152,6 +152,8 @@ if (UNIX)
 		set (MOSQ_LIBS ${MOSQ_LIBS} dl m)
     elseif (${CMAKE_SYSTEM_NAME} MATCHES "OpenBSD")
 	    set (MOSQ_LIBS ${MOSQ_LIBS} m)
+    elseif (${CMAKE_SYSTEM_NAME} MATCHES "NetBSD")
+	    set (MOSQ_LIBS ${MOSQ_LIBS} m)
     elseif(QNX)
         set(MOSQ_LIBS ${MOSQ_LIBS} m socket)
     else(APPLE)
@@ -185,9 +187,9 @@ target_link_libraries(mosquitto ${MOSQ_L
 if (UNIX)
 	if (APPLE)
 		set_target_properties(mosquitto PROPERTIES LINK_FLAGS "-Wl,-exported_symbols_list -Wl,${mosquitto_SOURCE_DIR}/src/linker-macosx.syms")
-	else (APPLE)
+	elseif (NOT CMAKE_SYSTEM_NAME STREQUAL SunOS)
 		set_target_properties(mosquitto PROPERTIES LINK_FLAGS "-Wl,-dynamic-list=${mosquitto_SOURCE_DIR}/src/linker.syms")
-	endif (APPLE)
+	endif ()
 endif (UNIX)
 
 install(TARGETS mosquitto RUNTIME DESTINATION "${CMAKE_INSTALL_SBINDIR}")
